### PR TITLE
[ADD] account_invoice_report_due_list: New module

### PR DESCRIPTION
**Pull request by carlosdauden**
@Tecnativa

### CODE REVIEW

File: account_invoice_report_due_list/README.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
+.. image:: https://img.shields.io/badge/license-AGPL--3-blue.png
+   :target: https://www.gnu.org/licenses/agpl
+   :alt: License: AGPL-3
+
+===============================
+Account Invoice Report Due List
+===============================
+
+This module extends the functionality of account to show multiple due data
```
Comment (pedrobaeza) [inline account_invoice_report_due_list/README.rst]: This module extends the invoice report for adding information about the corresponding due dates and amounts.

File: account_invoice_report_due_list/README.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+in invoice.
+
+Usage
+=====
+
+To use this module, you need to:
+
+#. Go to Account > Invoices
```
Comment (pedrobaeza) [inline account_invoice_report_due_list/README.rst]: Does this exist?

File: account_invoice_report_due_list/README.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+#. Select or create an invoice with a multiple lines payment term
```
Comment (pedrobaeza) [inline account_invoice_report_due_list/README.rst]: Select or create a payment term with multiple term lines.

File: account_invoice_report_due_list/i18n/es.po [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Plural-Forms: \n"
+"X-Generator: Poedit 1.8.7.1\n"
+
+#. module: account_invoice_report_due_list
+#: model:ir.model.fields,field_description:account_invoice_report_due_list.field_account_invoice_multi_date_due
+msgid "Due Dates"
+msgstr "Fechas vencimiento"
```
Comment (pedrobaeza) [inline account_invoice_report_due_list/i18n/es.po]: `Fechas de vencimiento`

File: account_invoice_report_due_list/i18n/es.po [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+#. module: account_invoice_report_due_list
+#: model:ir.model,name:account_invoice_report_due_list.model_account_invoice
+msgid "Invoice"
+msgstr "Factura"
+
+#. module: account_invoice_report_due_list
+#: model:ir.model.fields,field_description:account_invoice_report_due_list.field_account_invoice_multi_due
+msgid "Multiple date due"
+msgstr "Fecha vencimiento múlitiple"
```
Comment (pedrobaeza) [inline account_invoice_report_due_list/i18n/es.po]: Múltiples fechas de vencimiento

File: account_invoice_report_due_list/README.rst [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+Images
+------
+
+* Odoo Community Association: `Icon <https://odoo-community.org/logo.png>`_.
+
+Contributors
+------------
+
+* `Tecnativa <https://www.tecnativa.com>`_:
+  * Carlos Dauden
```
Comment (pedrobaeza) [inline account_invoice_report_due_list/README.rst]: Add line break in between for correct RST syntax

File: account_invoice_report_due_list/models/account_invoice.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+            invoice.multi_date_due = ' '.join(
+                fields.Date.from_string(due[0]).strftime(date_format)
+                for due in self.get_multi_due_list())
+
+    def get_multi_due_list(self):
+        self.ensure_one()
+        due_list = []
+        if self.move_id:
+            due_move_line_ids = self.move_id.line_ids.filtered(
+                lambda ml: ml.account_id.id == self.account_id.id
```
Comment (pedrobaeza) [inline account_invoice_report_due_list/models/account_invoice.py]: You shouldn't filter this way (and `.id` part is  useless), but filter the same as Odoo does: by account type according invoice type (in_*: payable, out_*: receivable).

File: account_invoice_report_due_list/views/account_invoice_view.xml [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    <field name="model">account.invoice</field>
+    <field name="inherit_id" ref="account.invoice_supplier_form"/>
+    <field name="arch" type="xml">
+        <field name="date_due" position="before">
+            <field name="multi_due" invisible="1"/>
+            <field name="multi_date_due" attrs="{'invisible': [('multi_due', '=', False)]}"/>
+        </field>
+        <field name="date_due" position="attributes">
+            <attribute name="attrs">{'invisible': [('multi_due', '=', True)]}</attribute>
+            <attribute name="groups"/>
```
Comment (pedrobaeza) [inline account_invoice_report_due_list/views/account_invoice_view.xml:16]: Why resetting groups attribute?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Add in the roadmap to add tests

**Comment by pedrobaeza** [comment]
@carlosdauden please check my comments

**Comment by feketemihai** [comment]
@carlosdauden @pedrobaeza will you take in consideration the amounts that need to be paid on dates, it will be a good improvement, something like in https://github.com/OCA/account-invoicing/pull/348 but more accurate since that pr didn't threat Invoicing in different currency..

**Comment by pedrobaeza** [comment]
@feketemihai indeed this doesn't take into account if the invoice is in other currency.

@carlosdauden please change the PR for using `amount_currency` instead of balance when `currency_id` of the account.move.line is different from `invoice_id.company_id.currency_id`.

**Comment by pedrobaeza** [comment]
@carlosdauden please finish this

**Comment by rafaelbn** [comment]
This PR has been in customer  production environment for 8 months without any issue I think is not WIP

**Review by rafaelbn — APPROVED** [review]
Functionally speaking :+1:

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**